src-electron/db/query-loader.js

```diff
@@ -938,54 +938,80 @@ async function insertAtomics(db, packageId, data) {
  * @param {*} context - The context containing the mandatory device type to check against.
  * @returns A promise resolved with the result of the database insert operation.
  */
-function insertEndpointComposition(db, composition, context) {
-  if (parseInt(context.mandatoryDeviceTypes, 16) === composition.code) {
-    return dbApi.dbInsert(
-      db,
-      'INSERT INTO ENDPOINT_COMPOSITION (TYPE, CODE) VALUES (?, ?)',
-      [dbEnum.composition.mandatoryEndpoint, composition.code],
-    )
-  } else {
-    return dbApi.dbInsert(
-      db,
-      'INSERT INTO ENDPOINT_COMPOSITION (TYPE, CODE) VALUES (?, ?)',
-      [composition.compositionType, composition.code],
-    )
+async function insertEndpointComposition(db, composition, context) {
+  try {
+    if (parseInt(context.mandatoryDeviceTypes, 16) === composition.code) {
+      return await dbApi.dbInsert(
```

Comment:
You do not do `return await xxxx` if this is an async function. You just retun the async itself.

Reply:
Ok I'll fix that